metaflow/plugins/aws/eks/kubernetes_client.py

```diff
@@ -0,0 +1,608 @@
+import os
+
+try:
+    unicode
+except NameError:
+    unicode = str
+    basestring = str
+
+from metaflow.exception import MetaflowException
+
+
+class KubernetesJobException(MetaflowException):
+    headline = "Kubernetes job error"
+
+
+class KubernetesClient(object):
+    def __init__(self):
+        # TODO: Look into removing the usage of Kubernetes Python SDK
+        # at some point in the future. Given that Kubernetes Python SDK
+        # aggressively drops support for older kubernetes clusters, continued
+        # dependency on it may bite our users.
+
+        try:
+            # Kubernetes is a soft dependency.
+            from kubernetes import client, config
+        except (NameError, ImportError):
+            raise MetaflowException(
+                "Could not import module 'kubernetes'. Install kubernetes "
+                "Python package (https://pypi.org/project/kubernetes/) first."
+            )
+        if os.getenv("KUBERNETES_SERVICE_HOST"):
+            # We’re inside a pod, authenticate via ServiceAccount assigned to us
+            config.load_incluster_config()
+        else:
+            # Use kubeconfig, likely $HOME/.kube/config
+            # TODO (savin):
+            #     1. Support generating kubeconfig on the fly using boto3
+            #     2. Support auth via OIDC - https://docs.aws.amazon.com/eks/latest/userguide/authenticate-oidc-identity-provider.html
+            # Supporting the above auth mechanisms (atleast 1.) should be
+            # good enough for the initial rollout.
+            config.load_kube_config()
+        self._client = client
+
+    def job(self, **kwargs):
+        return KubernetesJob(self._client, **kwargs)
+
+
+class KubernetesJob(object):
+    def __init__(self, client, **kwargs):
+        self._client = client
+        self._kwargs = kwargs
+
+        # Kubernetes namespace defaults to `default`
+        self._kwargs["namespace"] = self._kwargs["namespace"] or "default"
```

Comment:
This will fail if "namespace" is not in kwargs, need to use 
`   self._kwargs.get("namespace", "default")`